### ISSUE TITLE
Don't allow blank NS records to be saved

### ISSUE THREAD

**Issue by katypies** [cisagov/manage.get.gov#2980]
### Current Behavior

It's currently possible to save a blank NS record. This in turn results in a domain remaining in the "DNS Needed" state, even if they have other valid NS records.

### Expected Behavior

- Blank NS records shouldn't be saved; if a user tries to save one, we should either show an error, or remove the blank one.

### Steps to Reproduce

1. Go to a newly approved domain
2. Click to add NS records
3. Add only 2 records, but leave the third one blank
4. Save (blank can be saved, no error message)
5. Domain will remain in DNS needed even if the first two NS records are valid.

### Additional Context

By default, we're showing 3 blank record fields when a user initially sets up their nameservers. We should only show 2 (a primary and secondary) -- that way, users who have only 2 aren't forced to delete a default blank record, but users who have more than 2 can still add more if needed (which is a more obvious action than having to delete a blank/unused record).

[Slack thread with screenshots](https://cisa-corp.slack.com/archives/C06VDEWTGSH/p1729103344162159)

**Comment by h-m-f-t** [comment]
Putting this in our secret new dev parking lot status so it can get pointed sooner before moving to ready. @lizpearl @abroddrick

**Comment by katypies** [comment]
I'm going to move this one out. I tried to replicate this to get a bit more granular details, but I wasn't able to. Right now, my hunch is this might _only_ apply to domains that haven't ever entered the Ready state, but I want to do some additional testing to verify or get details to be sure it can be pointed correctly.

**Comment by abroddrick** [comment]
@katypies you mentioned moving this out, were you having trouble reproducing it? (Liz and I were trying to reproduce it and haven't been successful yet). The working theory is it's not the blank that caused this issue but something else. This may also just be resolved by the nameserver refactor we are doing in 62.

**Comment by katypies** [comment]
@abroddrick yes, I was having trouble reproducing. But, I just tried again, and think I may have been able to do it (save a blank record) on Staging.

Specifically, 
1. Went to a domain I can manage that was in the "DNS Needed" state (https://getgov-staging.app.cloud.gov/domain/13181/)
2. Clicked the option to add nameservers
3. Added two records, left the third blank
4. Clicked save

The blank record appeared to be saved. Even if I leave the DNS page, and return to edit the nameservers, there's a blank third record there.

The domain status shows as "Ready" so I'm not 100% sure this is the same issue as with the original report from the user (but also, I know Staging doesn't actually update any records -- so maybe the issue isn't the blank record being saved, but something if the blank record is transmitted to the registry, which causes an error?).

**Comment by gabydisarli** [comment]
@katypies This problem will be solved once the new name servers design is implemented which Dave, Rachid, and Aditi are working on this sprint.
